Output be more detailed for docker build cmd

- iotedgedev Version: 0.82
- Python Version: 3.6
- Pip Version:
- Development machine OS Version: Windows 10

Steps to Reproduce:
There's something wrong happened when building docker image, but the error message is too short.
code sample: https://github.com/VSChina/iot-edge-sample-solution/tree/priv-nuget

### Error message in iotedgedev
```
PS>iotedgedev push

=======================================
======== ENVIRONMENT VARIABLES ========
=======================================

Environment Variables loaded from: .env (C:\Users\v-zhq\IoT\edge\newsolution\edge-multi-module\.env)

==================================
======== BUILDING MODULES ========
==================================

BUILDING MODULE: FilterModule
PROCESSING DOCKERFILE: C:\Users\v-zhq\IoT\edge\newsolution\edge-multi-module\modules\FilterModule\Dockerfile
BUILDING DOCKER IMAGE: michaeljqzq/multi-1:0.0.1-amd64
ERROR: Error: The command '/bin/sh -c dotnet restore' returned a non-zero code: 1
```

### Error message in docker build
```
PS>docker build -t "temp" .
Sending build context to Docker daemon  18.43kB
Step 1/10 : FROM microsoft/dotnet:2.0-sdk AS build-env
 ---> 6701b02eaaeb
Step 2/10 : WORKDIR /app
 ---> Using cache
 ---> 0172620f7208
Step 3/10 : COPY *.csproj ./
 ---> 9f5f39d157ce
Step 4/10 : RUN dotnet restore
 ---> Running in 4ef88597435d
  Restoring packages for /app/FilterModule.csproj...
  Installing Microsoft.Extensions.DependencyInjection.Abstractions 1.1.0.
  Installing Microsoft.Extensions.Logging.Abstractions 1.1.1.
  Installing Microsoft.Extensions.Logging 1.1.1.
  Installing System.Net.NetworkInformation 4.3.0.
  Installing DotNetty.Common 0.4.6.
  Installing DotNetty.Codecs 0.4.6.
  Installing DotNetty.Transport 0.4.6.
  Installing DotNetty.Buffers 0.4.6.
  Installing Newtonsoft.Json 10.0.3.
  Installing System.Net.WebSockets 4.3.0.
  Installing System.IO.FileSystem.Watcher 4.3.0.
  Installing Microsoft.Extensions.FileSystemGlobbing 2.0.0-preview2-final.
  Installing Microsoft.Extensions.FileProviders.Abstractions 2.0.0-preview2-final.
  Installing System.Runtime.CompilerServices.Unsafe 4.4.0-preview2-25405-01.
  Installing Microsoft.Packaging.Tools 1.0.0-preview2-25401-01.
  Installing DotNetty.Handlers 0.4.6.
  Installing DotNetty.Codecs.Mqtt 0.4.6.
  Installing Microsoft.Azure.Devices.Shared 1.2.0-preview-001.
  Installing System.Net.WebSockets.Client 4.3.0.
  Installing WindowsAzure.Storage 8.1.1.
  Installing System.Net.Requests 4.3.0.
  Installing System.Net.Http 4.3.1.
  Installing Microsoft.Azure.Amqp 2.0.6.
  Installing Validation 2.4.13.
  Installing System.Threading.Overlapped 4.3.0.
  Installing System.Net.WebHeaderCollection 4.3.0.
  Installing System.Diagnostics.TraceSource 4.3.0.
  Installing Microsoft.Extensions.FileProviders.Physical 2.0.0-preview2-final.
  Installing Microsoft.Extensions.Primitives 2.0.0-preview2-final.
  Installing NETStandard.Library 2.0.0-preview2-25401-01.
  Installing Microsoft.Azure.Devices.Client 1.6.0-preview-001.
  Installing Microsoft.Extensions.Configuration.FileExtensions 2.0.0-preview2-final.
  Installing Microsoft.Extensions.Configuration.Abstractions 2.0.0-preview2-final.
  Installing Microsoft.Extensions.Configuration.Binder 2.0.0-preview2-final.
  Installing Microsoft.Extensions.Configuration.Json 2.0.0-preview2-final.
  Installing Microsoft.Extensions.Configuration.EnvironmentVariables 2.0.0-preview2-final.
  Installing Microsoft.Extensions.Configuration 2.0.0-preview2-final.
  Installing System.Runtime.Loader 4.3.0.
/app/FilterModule.csproj : error NU1101: Unable to find package Zhiqing.hello. No packages exist with this id in source(s): nuget.org
  Generating MSBuild file /app/obj/FilterModule.csproj.nuget.g.props.
  Generating MSBuild file /app/obj/FilterModule.csproj.nuget.g.targets.
  Restore failed in 6.89 sec for /app/FilterModule.csproj.
The command '/bin/sh -c dotnet restore' returned a non-zero code: 1
```

--- a/iotedgedev/dockercls.py
+++ b/iotedgedev/dockercls.py
@@ -1,8 +1,8 @@
-import json
 import os
 import zipfile
 
 import docker
+import six
 
 from .moduletype import ModuleType
 
@@ -233,16 +233,13 @@ class Docker:
         self.output.info("Log files successfully saved to: " + zip_path)
 
     def process_api_response(self, response):
-        for line in response:
-            decoded = self.utility.decode(line).replace('\\u003e', '>')
-            self.output.procout(decoded)
-            try:
-                decoded_json = json.loads(decoded)
-            except ValueError:
-                continue
+        for json_ in docker.utils.json_stream.json_stream(response):
+            for key in json_:
+                if key in {"status", "stream"} and isinstance(json_[key], six.string_types):
+                    self.output.procout(json_[key].rstrip())
 
             # Docker SDK won't throw exceptions for some failures.
             # We have to check the response ourselves.
             # Related issue: https://github.com/docker/docker-py/issues/1772
-            if 'error' in decoded_json:
-                raise ValueError(decoded_json['error'])
+            if "error" in json_:
+                raise ValueError(json_["error"])

--- a/iotedgedev/modules.py
+++ b/iotedgedev/modules.py
@@ -181,10 +181,11 @@ class Modules:
                     if docker.get_os_type() == "linux" and sys.platform == "win32":
                         dockerfile_relative = dockerfile_relative.replace("\\", "/")
 
-                    build_result = docker.docker_client.images.build(tag=tag, path=context_path, dockerfile=dockerfile_relative, **sdk_options)
+                    build_args = {"tag": tag, "path": context_path, "dockerfile": dockerfile_relative}
+                    build_args.update(sdk_options)
 
-                    self.output.info("DOCKER IMAGE DETAILS: {0}".format(build_result))
-
+                    response = docker.docker_api.build(**build_args)
+                    docker.process_api_response(response)
                 if not no_push:
                     docker.init_registry()
 

